put/request.go

```diff
@@ -622,3 +628,59 @@ func handleNamespace(key string) (string, error) {
 		return "", errInvalidMetaNamespace
 	}
 }
+
+// ValidateAndCreateReasonMetadata returns an error if the provided string is
+// not a valid ibackup reason, otherwise returns the related reason metadata
+// key.
+func ValidateAndCreateReasonMetadata(reason string) (string, error) {
```

Comment:
Nitpicking slightly, but if this isn't moved to a custom Flag type, it'd probably be better (read: more performant) as a `switch` statement. The performance doesn't matter that much, especially as it's a small func that is only ran a single time, but it's good to get in the habit.